packages/client/src/authentication/Authentication.ts

```diff
@@ -0,0 +1,43 @@
+import { invariant } from '@lens-protocol/shared-kernel';
+import { GraphQLClient } from 'graphql-request';
+
+import { getSdk, Sdk } from './graphql/auth.generated';
+
+export type Credentials = { accessToken: string; refreshToken: string };
+
+export class Authentication {
+  private readonly sdk: Sdk;
+
+  constructor(backendUrl: string) {
```

Comment:
nit: for consistency..

I think throughout the codebase we use `backendUrl` to identify the URL of the API. Then the API includes a `/graphql` endpoint and other endpoints that are not GQL.. we will have more for direct file uploads and so on.

What if we rename all these as:
```suggestion
  constructor(gqlEndpoint: string) {
```
and redefine `Environment` as:
```
type Environment = {
  url: string;
  
  get gqlEndpoint(): string
};
```

or simpler with a class as:
```ts
class Environment = {
  constructor(private url: string) {}
  
  get gqlEndpoint() {
    return `${this.url}/graphql`
  }
};
```

Thoughts?

Reply:
I did it as you suggested, thanks for the idea!

Reply:
> what about someone having some proxies that rewrite url to be just `foo.xyz` (or even `foo.xyz/lens-api)?

not sure I follow @desfero , can you please clarify the scenario?

Reply:
@cesarenaldi the interface to provide the `environment` is public. We can't stop anyone from having a proxy in between `client` and our `graphql api` (for .e.g someone wants to have custom metrics around perfomance, api usage by using SAAS similar to Stellate). The proxy may be configured in a way that doesn't expose graphql under `<url>/graphql` so we should allow passing the full graphql url without appending anything on our side.
Any thoughts? Does that make sense anyway?

Reply:
I think it makes sense so I won't append anything to the API URL